### PR TITLE
deployment: add limits to k8s-kvm image

### PR DESCRIPTION
**Pull request by kopiczko**
Towards https://github.com/giantswarm/giantswarm/issues/1173

I added the limits to the `k8s-kvm` container. The rest is left as it is.

### CODE REVIEW

File: service/key/key.go [hunk 1]
```diff
@@ -38,6 +40,15 @@ func ConfigMapName(customObject kvmtpr.CustomObject, node spec.Node, prefix stri
 	return fmt.Sprintf("%s-%s-%s", prefix, ClusterID(customObject), node.ID)
 }
 
+func CPUQuanity(n kvmtprkvm.Node) (resource.Quantity, error) {
```
Comment (xh3b4sd) [inline service/key/key.go]: It is `Quantity`.

**Comment by kopiczko** [comment]
I hope qemu overhead is small comparing to 1G. Please don't make me parse those and add something :)

**Comment by JosephSalisbury** [comment]
What I would suggest is trying this out, and then seeing what happens when the memory on a machine is really high - I'm a bit worried that kubernetes will kill it due to OOM

File: service/key/key.go (hunk 1 above)
Comment (JosephSalisbury) [reply service/key/key.go]: I can't believe I missed this :D

**Comment by JosephSalisbury** [comment]
Me and @kopiczko chatted. I didn't notice that this was only requests, not limits, so we won't kill a machine that uses a lot of memory (which is fine, because qemu handles our cpu / memory foo). All good.

File: service/key/key.go (hunk 1 above)
Comment (kopiczko) [reply service/key/key.go]: Done.

**Comment by kopiczko** [comment]
Tested on lycan.
I wait for SRE approval.

File: service/resource/deployment/worker_deployment.go [hunk 2]
```diff
@@ -22,7 +22,7 @@ func newWorkerDeployments(customObject kvmtpr.CustomObject) ([]*extensionsv1.Dep
 	for i, workerNode := range customObject.Spec.Cluster.Workers {
 		capabilities := customObject.Spec.KVM.Workers[i]
 
-		cpuQuanity, err := key.CPUQuanity(capabilities)
+		cpuQuanity, err := key.CPUQuantity(capabilities)
```
Comment (corest) [inline service/resource/deployment/worker_deployment.go]: What about var name? It is also wrong

File: service/resource/deployment/master_deployment.go [hunk 3]
```diff
@@ -22,7 +22,7 @@ func newMasterDeployments(customObject kvmtpr.CustomObject) ([]*extensionsv1.Dep
 	for i, masterNode := range customObject.Spec.Cluster.Masters {
 		capabilities := customObject.Spec.KVM.Masters[i]
 
-		cpuQuanity, err := key.CPUQuanity(capabilities)
+		cpuQuanity, err := key.CPUQuantity(capabilities)
```
Comment (corest) [inline service/resource/deployment/master_deployment.go]: What about var names?

File: service/resource/deployment/worker_deployment.go (hunk 2 above)
Comment (kopiczko) [reply service/resource/deployment/worker_deployment.go]:
```
for file in $(find . -type d \( -path ./.git -o -path ./vendor \)  -prune -o -print | grep '.*\.go'); do
    sed -i -e 's/Quanity/Quantity/g' $file
done
```

Comment (kopiczko) [reply service/resource/deployment/worker_deployment.go]: Done, thanks :)

Comment (xh3b4sd) [reply service/resource/deployment/worker_deployment.go]: You should really get a decent text editor. :trollface:

Comment (kopiczko) [reply service/resource/deployment/worker_deployment.go]: My editor is capable for doing renaming, but it's too sloooow.

**Pull request merged**